packages/components/modules/payments/web/CheckoutComponent/Checkout/index.tsx

```diff
@@ -0,0 +1,281 @@
+          },
+          onError: (error: any) => {
+            const message = extractErrorMessage(error)
+            sendToast(message, { type: 'error' })
+            setIsRetry(true)
+          },
+        },
+      )
+    } catch (error: any) {
+      const message = error?.response?.data?.error || error?.message || 'Please try again.'
+      sendToast(`Failed to create subscription: ${message}`, { type: 'error' })
+      setIsRetry(true)
+    }
+  }
+
+  const handleRetry = () => {
+    setIsRetry(false)
+    sendToast('Retrying subscription creation...', { type: 'info' })
+    handlePlaceOrder()
+  }
+
+  const addressOptions = {
+    mode: 'billing' as const,
+    ...{
+      defaultValues: {
+        name: selectedMethod?.billingDetails?.name || '',
+        address: {
+          line1: selectedMethod?.billingDetails?.address?.line1 || '',
+          line2: selectedMethod?.billingDetails?.address?.line2 || '',
+          city: selectedMethod?.billingDetails?.address?.city || '',
+          state: selectedMethod?.billingDetails?.address?.state || '',
+          postal_code: selectedMethod?.billingDetails?.address?.postalCode || '',
+          country: selectedMethod?.billingDetails?.address?.country || 'US',
+        },
+      },
+    },
+  }
+
+  const shouldRenderAddressElement =
+    !isAddCardModalOpen && !isLoadingMethods && paymentMethods.length > 0
+
+  return (
+    <Box sx={{ width: '100%', padding: 2 }}>
+      <Grid
+        container
+        spacing={3}
+        direction={isMobile ? 'column' : 'row-reverse'}
+        justifyContent="center"
+      >
+        <Grid item xs={12} sm={6}>
+          <Box sx={{ maxWidth: 400 }}>
+            <ProductContainer>
+              <Box sx={{ display: 'flex', alignItems: 'center', gap: 2 }}>
+                {product?.images.length > 0 && (
+                  <Box
+                    component="img"
+                    src={product?.images[0]}
+                    alt={product?.name}
+                    sx={{ width: 48, height: 48, borderRadius: 1 }}
+                  />
+                )}
+                <Typography variant="body2" fontWeight={700}>
+                  {product?.name}
+                </Typography>
+              </Box>
+              <Box>
+                {Number.isFinite(product?.defaultPrice?.unitAmount) && (
+                  <Box sx={{ display: 'flex', justifyContent: 'flex-end', gap: 1 }}>
+                    <Typography variant="body2" fontWeight={700}>
+                      {(product.defaultPrice.unitAmount / 100).toLocaleString('en-US', {
+                        style: 'currency',
+                        currency: 'USD',
+                      })}
+                    </Typography>
+                  </Box>
+                )}
+                <Typography variant="body2" color="text.secondary">
+                  + taxes /month
+                </Typography>
+              </Box>
+            </ProductContainer>
+            <Box sx={{ mt: 2 }}>
```

Comment:
Maybe use gap on the parent component instead of margin here and on the box bellow